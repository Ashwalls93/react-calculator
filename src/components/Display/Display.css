.display-container {
  align-items: center;
  display: flex;
  height: var(--display-height);
  padding: 0 4%;
  transition: var(--default-transition);
  width: var(--display-width);
}

.display-theme-default {
  background-color: var(--display-background-color-default);
}

.display-theme-pony {
  background-color: var(--display-background-color-pony);
  border-radius: 8px 8px 0 0;
}

.display-theme-dondi {
  background-color: var(--display-background-color-dondi);
  border-radius: 8px 8px 0 0;
}

.display-value {
  font-size: var(--display-text-size);
  font-family: var(--main-font);
  font-weight: var(--display-text-weight);
  margin-left: auto;
  overflow: hidden;
}

.display-default {
  color: var(--display-text-color-default);
}

.display-pony {
  color: var(--display-text-color-pony);
}

.display-dondi {
  color: var(--display-text-color-dondi);
}