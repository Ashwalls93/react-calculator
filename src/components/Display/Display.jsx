import React from 'react';
import PropTypes from 'prop-types';
import './Display.css';

const Display = ({theme, displayValue}) => {
  return (
    <div className={`display-container display-theme-${theme}`}>
      <p className={`display-value display-${theme}`}>
        {displayValue}
      </p>
    </div>
  );
}

Display.propTypes = {
  displayValue: PropTypes.string.isRequired,
  theme: PropTypes.string.isRequired,
};

export default Display;