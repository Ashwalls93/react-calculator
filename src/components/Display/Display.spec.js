import React from 'react';
import {mount, shallow} from 'enzyme';
import Display from './Display';

describe('Display', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Display 
        displayValue={''} 
        theme={'default'}
        toggleTheme={jest.fn()}
      />
    );
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  it('should render a <div />', () => {
    expect(wrapper.find('div').length).toEqual(1);
  });

  it('renders the value of displayValue', () => {
    wrapper.setProps({displayValue: 'test'});
    expect(wrapper.find('.display-value').text()).toEqual('test');;
  });
    
  it('should have a className reflecting its theme', () => {
    expect(wrapper.find('.display-theme-default').exists()).toBe(true);
  });

  it('should update its className on theme change', () => {
    expect(wrapper.find('.display-theme-default').exists()).toBe(true);
    wrapper.setProps({theme: 'pony'});
    expect(wrapper.find('.display-theme-pony').exists()).toBe(true);
  });
});