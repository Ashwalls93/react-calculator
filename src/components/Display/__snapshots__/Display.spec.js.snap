// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Display
    displayValue=""
    theme="default"
    toggleTheme={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p
        className="display-value display-default"
/>,
      "className": "display-container display-theme-default",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "",
        "className": "display-value display-default",
      },
      "ref": null,
      "rendered": "",
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p
          className="display-value display-default"
/>,
        "className": "display-container display-theme-default",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "display-value display-default",
        },
        "ref": null,
        "rendered": "",
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
